src/BaseSelect.tsx

```diff
@@ -691,6 +691,10 @@ const BaseSelect = React.forwardRef((props: BaseSelectProps, ref: React.Ref<Base
   const onClearMouseDown: React.MouseEventHandler<HTMLSpanElement> = () => {
     onClear?.();
 
+    selectorRef.current?.focus();
+    setMockFocused(true);
```

Comment:
focus will trigger `onFocus` event which will call `setMockFocused(true)`, no need manually call here and follow code also should be auto.